### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by riazarbi**
In `rstudio_shiny` subdirectory - 

1. Moved apt.sh into Dockerfile to allow for layered builds
2. Updated the RStudio and Shiny debs to latest
3. Moved the Dockerfile contents around so that the long-running stuff is at the top, speeding up builds that just involve modifications to conf files
4. Fixed sed modification of nginx conf files to only occur if VIRTUAL_PATH has been changed from `/`
5. Fixed shiny nginx conf - it was redirecting to root url.

### CODE REVIEW

**Comment by riazarbi** [comment]
Everything relating to Rstudio and Shiny works with this branch at present. Suggest we resolve merges by always favouring this branch over staged_builds branch

File: base/python/R/R_nonstandard/rstudio_shiny/Dockerfile [hunk 1]
```diff
@@ -2,21 +2,56 @@ FROM cityofcapetown/datascience:R_nonstandard
...
+COPY run_rstudio.sh /run_scripts/
+RUN chmod +x /run_scripts/run_rstudio.sh
 COPY run_shiny.sh /run_scripts/
 RUN chmod +x /run_scripts/run_shiny.sh
 
+# Copy in the NGINX confs
+COPY rstudio.conf /etc/nginx/paths-available/
 COPY shiny.conf /etc/nginx/paths-available/
+
+
```
Comment (Gordonei) [inline base/python/R/R_nonstandard/rstudio_shiny/Dockerfile:55]: Remove blank lines at end of file?

File: base/python/R/R_nonstandard/rstudio_shiny/Dockerfile [hunk 2]
```diff
@@ -2,21 +2,56 @@ FROM cityofcapetown/datascience:R_nonstandard
 
 LABEL authors="Riaz Arbi,Gordon Inggs"
 
-# Install rstudio
-COPY rstudio_additions.sh .
-RUN bash rstudio_additions.sh
+# Install and setup Pandoc
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
```
Comment (Gordonei) [inline base/python/R/R_nonstandard/rstudio_shiny/Dockerfile:6]: I subsequently did this in a commit to `staged_builds`. 

Please rebase against it - based on the merge conflicts, it should be reasonably seamless.

File: base/python/R/R_nonstandard/rstudio_shiny/run_rstudio.sh [hunk 3]
```diff
@@ -1,8 +1,14 @@
 #!/usr/bin/env bash
 
-# Fixing RStudio redirect
-sed -i "\/proxy_redirect http\:\/\/localhost\:8787\//c\proxy_redirect http\:\/\/localhost\:8787\/ http\:\/\/\$host$VIRTUAL_PATH\/rstudio\/\;" /etc/nginx/paths-available/rstudio.conf
-sed -i "\/proxy_redirect https\:\/\/localhost\:8787\//c\proxy_redirect https\:\/\/localhost\:8787\/ https\:\/\/\$host$VIRTUAL_PATH\/rstudio\/\;" /etc/nginx/paths-available/rstudio.conf
+# Fix NGINX hostname for dynamic routing
+if [[ $VIRTUAL_PATH = "/" ]]; then
```
Comment (Gordonei) [inline base/python/R/R_nonstandard/rstudio_shiny/run_rstudio.sh]: Ditto for this - please rebase against `staged_builds`.

File: base/python/R/R_nonstandard/rstudio_shiny/run_shiny.sh [hunk 4]
```diff
@@ -1,9 +1,14 @@
 #!/usr/bin/env bash
 
-# Fixing Shiny redirect
-sed -i "\/rewrite \^\/shiny\$/c\/rewrite \^\/shiny\$ \$scheme\:\/\/\$http_host$VIRTUAL_PATH\/shiny\/ permanent;"
-sed -i "\/proxy_redirect http\:\/\/localhost\:3838\//c\proxy_redirect http\:\/\/localhost\:3838\/ http\:\/\/\$host$VIRTUAL_PATH\/shiny\/\;" /etc/nginx/paths-available/shiny.conf
-sed -i "\/proxy_redirect https\:\/\/localhost\:3838\//c\proxy_redirect https\:\/\/localhost\:3838\/ https\:\/\/\$host$VIRTUAL_PATH\/shiny\/\;" /etc/nginx/paths-available/shiny.conf
+if [[ $VIRTUAL_PATH = "/" ]]; then
```
Comment (Gordonei) [inline base/python/R/R_nonstandard/rstudio_shiny/run_shiny.sh]: As before, please rebase.

**Review by Gordonei — COMMENTED** [review]
Mostly just rebasing.

File: base/python/R/R_nonstandard/rstudio_shiny/shiny-server-standalone.conf [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Define the user we should use when spawning R Shiny processes
```
Comment (Gordonei) [inline base/python/R/R_nonstandard/rstudio_shiny/shiny-server-standalone.conf]: This is useful, but should rather be in its own image, depending on `R_nonstandard`. Please park in another branch until we do that.

Comment (riazarbi) [reply base/python/R/R_nonstandard/rstudio_shiny/shiny-server-standalone.conf]: Done!

**Pull request merged**